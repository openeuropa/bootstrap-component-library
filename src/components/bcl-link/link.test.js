import { renderTwigFileAsNode, getVariants } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-data-link/data";
import toggleDemoData from "@openeuropa/bcl-data-link/toggleData";
import tooltipDemoData from "@openeuropa/bcl-data-link/tooltipData";

const template = "@oe-bcl/bcl-link/link.html.twig";
const render = (params, reset) => renderTwigFileAsNode(template, params, reset);
const variants = getVariants();

describe("OE - Link", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoData, true)).resolves.toMatchSnapshot();
  });

  test(`with underline hover class renders correctly`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, standalone: true })
    ).resolves.toMatchSnapshot();
  });

  variants.forEach((variant) => {
    test(`${variant} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render({ ...demoData, variant: variant }, true)
      ).resolves.toMatchSnapshot();
    });
  });

  test(`as toggle renders correctly`, () => {
    expect.assertions(1);

    return expect(render(toggleDemoData)).resolves.toMatchSnapshot();
  });

  test(`with tooltip renders correctly`, () => {
    expect.assertions(1);

    return expect(render(tooltipDemoData)).resolves.toMatchSnapshot();
  });

  test("with icon renders correctly", () => {
    expect.assertions(1);

    return expect(
      render(
        {
          ...demoData,
          icon: {
            name: "mouse",
            transformation: "rotate-90",
            path: "/icons.svg",
          },
        },
        true
      )
    ).resolves.toMatchSnapshot();
  });
});
