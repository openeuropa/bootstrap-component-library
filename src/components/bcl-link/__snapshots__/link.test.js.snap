// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Link as toggle renders correctly 1`] = `
<jest>
  <a
    aria-controls="collapseExample"
    aria-expanded="false"
    data-bs-toggle="collapse"
    href="#collapseExample"
    role="button"
  >
    I am a link
  </a>
</jest>
`;

exports[`OE - Link danger renders correctly 1`] = `
<jest>
  <a
    class="link-danger"
    href="/example.html"
  >
    A simple link!
  </a>
</jest>
`;

exports[`OE - Link dark renders correctly 1`] = `
<jest>
  <a
    class="link-dark"
    href="/example.html"
  >
    A simple link!
  </a>
</jest>
`;

exports[`OE - Link info renders correctly 1`] = `
<jest>
  <a
    class="link-info"
    href="/example.html"
  >
    A simple link!
  </a>
</jest>
`;

exports[`OE - Link light renders correctly 1`] = `
<jest>
  <a
    class="link-light"
    href="/example.html"
  >
    A simple link!
  </a>
</jest>
`;

exports[`OE - Link primary renders correctly 1`] = `
<jest>
  <a
    class="link-primary"
    href="/example.html"
  >
    A simple link!
  </a>
</jest>
`;

exports[`OE - Link renders correctly 1`] = `
<jest>
  <a
    href="/example.html"
  >
    A simple link!
  </a>
</jest>
`;

exports[`OE - Link secondary renders correctly 1`] = `
<jest>
  <a
    class="link-secondary"
    href="/example.html"
  >
    A simple link!
  </a>
</jest>
`;

exports[`OE - Link success renders correctly 1`] = `
<jest>
  <a
    class="link-success"
    href="/example.html"
  >
    A simple link!
  </a>
</jest>
`;

exports[`OE - Link warning renders correctly 1`] = `
<jest>
  <a
    class="link-warning"
    href="/example.html"
  >
    A simple link!
  </a>
</jest>
`;

exports[`OE - Link with icon renders correctly 1`] = `
<jest>
  <a
    href="/example.html"
  >
    A simple link!
    <svg
      class="bi icon--s icon--rotate-90"
    >
      <use
        xlink:href="bootstrap-icons.svg#mouse"
      />
    </svg>
  </a>
</jest>
`;

exports[`OE - Link with tooltip renders correctly 1`] = `
<jest>
  <a
    data-bs-toggle="tooltip"
    href="/example.html"
    role="button"
    title="This is a tooltip"
  >
    A link with tooltip
  </a>
</jest>
`;
